src/client/sandbox/event/selection.js

```diff
@@ -26,8 +27,9 @@ export default class Selection {
             var isElementActive    = false;
 
             var selectionSetter = () => {
-                var changeType = Selection._needChangeInputType(el);
-                var savedType  = el.type;
+                var changeType           = Selection._needChangeInputType(el);
+                var useInternalSelection = Selection._needForInternalSelection();
```

Comment:
with this code if `input.type` is `text` and it's Firefox v51, the `useInternalSelection` var is true. But I suppose we should not use internal selection in this case...